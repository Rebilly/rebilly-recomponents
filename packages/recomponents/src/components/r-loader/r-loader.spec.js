import {shallowMount} from '@vue/test-utils';
import {renderToString} from '@vue/server-test-utils';
import RLoader from './r-loader.vue';

describe('r-loader.vue', () => {
    it('should render via SSR and match snapshot', async () => {
        const wrapper = renderToString(RLoader, {
            propsData: {
                show: true,
            },
        });

        expect(wrapper).toMatchSnapshot();
    });

    it('should not render if show property is false', () => {
        const wrapper = shallowMount(RLoader, {
            propsData: {
                show: false,
            },
        });

        expect(wrapper).toMatchSnapshot();
    });

    it('should render component with classes', () => {
        const wrapper = shallowMount(RLoader, {
            propsData: {
                show: true,
                loading: true,
                fullscreen: true,
            },
        });

        expect(wrapper.classes()).toContain('r-loader--is-loading');
        expect(wrapper.classes()).toContain('r-loader--fullscreen');
    });

    it('should match all incoming props types', () => {
        const {show, loading, fullscreen} = RLoader.props;

        expect(show.type).toBe(Boolean);
        expect(loading.type).toBe(Boolean);
        expect(fullscreen.type).toBe(Boolean);
    });
});
