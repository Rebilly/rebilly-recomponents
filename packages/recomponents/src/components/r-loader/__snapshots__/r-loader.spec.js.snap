// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`r-loader.vue should not render if show property is false 1`] = `
<div class="r-component r-loader r-loader--loading r-loader--fullscreen">
  <div class="r-loader-wrapper">
    <div class="r-mr-re"></div>
    <div class="r-mr-bil"></div>
    <div class="r-mr-ly"></div>
  </div>
</div>
`;

exports[`r-loader.vue should render via SSR and match snapshot 1`] = `
<div class="r-component r-loader r-loader--loading r-loader--fullscreen">
  <div class="r-loader-wrapper">
    <div class="r-mr-re"></div>
    <div class="r-mr-bil"></div>
    <div class="r-mr-ly"></div>
  </div>
</div>
`;
